src/main/dynamo/iterators/pagedIterator.ts

```diff
@@ -58,6 +58,7 @@ export async function* pagedIterator<T, U extends TaggedModel>(
         lastEvaluatedKey = response.LastEvaluatedKey
         pendingOperation = buildOperation({ ...options, lastEvaluatedKey })
       } else {
+        lastEvaluatedKey = undefined
```

Comment:
The root cause of the bug is we weren't setting our local copy of `lastEvaluatedKey` to `undefined` when Dynamo's response contained  `LastEvaluatedKey: undefined`